test/zdtm/static/maps11.c

```diff
@@ -0,0 +1,112 @@
+#include <stdint.h>
+#include <signal.h>
+#include <sys/mman.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+#include "zdtmtst.h"
+
+#ifndef MAP_DROPPABLE
+#define MAP_DROPPABLE 0x08
+#endif
+
+#ifndef MADV_WIPEONFORK
+#define MADV_WIPEONFORK 18
+#endif
+
+const char *test_doc = "Test MAP_DROPPABLE/MADV_WIPEONFORK mappings with 2 processes";
+const char *test_author = "Alexander Mikhalitsyn <aleksandr.mikhalitsyn@canonical.com>";
+
+int main(int argc, char **argv)
+{
+	uint8_t *p1, *p2;
+	task_waiter_t t;
+	pid_t pid;
+	int status;
+	const char data[] = "MADV_WIPEONFORK vma data";
+
+	test_init(argc, argv);
+	task_waiter_init(&t);
+
+	p1 = mmap(NULL, sizeof(data), PROT_READ | PROT_WRITE, MAP_DROPPABLE | MAP_ANONYMOUS, 0, 0);
+	if (p1 == MAP_FAILED) {
+		if (errno == EINVAL) {
+			skip("mmap failed, no kernel support for MAP_DROPPABLE\n");
+			goto skip;
+		} else {
+			pr_perror("mmap failed");
+			return -1;
+		}
+	}
+
+	p2 = mmap(NULL, sizeof(data), PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS, 0, 0);
+	if (p2 == MAP_FAILED) {
+		pr_perror("mmap failed");
+		return 1;
+	}
+
+	if (madvise(p2, sizeof(data), MADV_WIPEONFORK)) {
+		pr_perror("madvise failed");
+		return -1;
+	}
+
+	memcpy(p1, data, sizeof(data));
+	memcpy(p2, data, sizeof(data));
+
+	pid = test_fork();
```

Comment:
why do we do fork before C/R (test_waitsig)?

Reply:
I have added an explanatory comment in there and also second `fork()` after C/R to make test even more strict.

Thanks!